flask_praetorian/base.py

```diff
@@ -693,6 +833,32 @@ def get_user_from_registration_token(self, token):
         )
         return user
 
+    def validate_reset_token(self, token):
+        """
+        Validates a password reset request based on the reset token
+        that is supplied. Verifies that the token is a reset token
+        and that the user can be properly retrieved
+        """
+        data = self.extract_jwt_token(token, access_type=AccessType.reset)
+        flask.current_app.logger.debug("DATA: {}".format(data))
+        user_id = data.get('id')
+        PraetorianError.require_condition(
+            user_id is not None,
+            "Could not fetch an id from the reset token",
+        )
+        user = self.user_class.identify(user_id)
+        PraetorianError.require_condition(
+            user is not None,
+            "Could not identify the user from the reset token",
+        )
+        # TODO: this should probably just be mandatory to prevent replay
```

Comment:
I think having a short lifespan for the token can mitigate this risk. I don't want to add another requirement for the user store. If the user wants to add logic to make sure that a reset token is only used once, that should probably be handled in their app.

My concern is that adding more requirements to the user store makes flask-praetorian less adaptable/user-friendly, especially for an edge case that might not be needed

Reply:
I agree ... security hat wants to slap me, but for sanity sake with the new feature, best to error on the side of caution.  removed usages/tests.